internal/bootfiles/bootfiles.go

```diff
@@ -55,3 +60,30 @@ func (b *BootFiles) DownloadBootFiles(ctx context.Context, params operations.Dow
 	return filemiddleware.NewResponder(operations.NewDownloadBootFilesOK().WithPayload(reader),
 		objectName, contentLength)
 }
+
+func (b *BootFiles) UploadBootFiles(ctx context.Context, params operations.UploadBootFilesParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+
+	// Create/Update version in DB
+	version := *params.UploadBootFilesParams.OpenshiftVersion
+	tx := b.db.Begin()
+	if err := tx.Save(&version).Error; err != nil {
+		err = errors.Wrapf(err, "Error storing OCP and RHCOS versions in DB")
+		log.Error(err)
+		return common.GenerateErrorResponder(err)
+	}
+	if err := tx.Commit().Error; err != nil {
+		return common.GenerateErrorResponder(err)
+	}
+
+	// Upload relevant boot files according to the specified version
+	ocpVersion := *params.UploadBootFilesParams.OpenshiftVersion.DisplayName
+	haveLatestMinimalTemplate := s3wrapper.HaveLatestMinimalTemplate(ctx, log, b.objectHandler)
+	if err := b.objectHandler.UploadBootFiles(ctx, ocpVersion, b.serviceBaseURL, haveLatestMinimalTemplate); err != nil {
```

Comment:
This operation might take a while.. Is there a way to return a stream with the progress to the user?

Reply:
Moved to an internal api (InstallerInternals->AddOpenshiftVersion), keeping the operation as sync before creating the cluster.